tests/misc/llext-edk/pytest/test_edk.py

```diff
@@ -45,6 +47,12 @@ def test_edk(dut: DeviceAdapter):
             ext_dir = Path(os.environ["ZEPHYR_BASE"]) / "tests/misc/llext-edk/extension"
```

Comment:
to get source path you can try:
```
            ext_dir = Path(find_in_config(dut.device_config.build_dir / 'CMakeCache.txt',
                                          'APPLICATION_SOURCE_DIR:PATH')) / 'extension'
```

Reply:
That's a bit obscure, no? Maybe we should instead embrace ZEPHYR_BASE and have a fixture providing it? It seems (to me) more idiomatic in Zephyr. I saw one, but it's only for pytest-twister-harness own test.

Reply:
yes, a bit obscure ... like a lot of things we are doing :) It was just a hint, how to get a source dir path, when you have only a build dir provided, to not have to hardcode the source dir in a code. When you want to reuse that code in another test scenario (prepare common method), that better to read that.
Reading ZEPHYR_BASE from path is ok. You can use a fixture, it is a plus when you have many tests and checking that multiple times (with a fixture you do it only once).